terraform/production/service_account_roles.tf

```diff
@@ -77,72 +77,19 @@ resource "kubectl_manifest" "notary_service_account" {
   ]
 }
 
-# IAM Role for granting read access to spack-binaries and write access to the cache index
-data "aws_iam_policy_document" "cache_indexer_assume_role_policy" {
-  statement {
-    sid     = ""
-    effect  = "Allow"
-    actions = ["sts:AssumeRoleWithWebIdentity"]
-
-    condition {
-      test     = "StringEquals"
-      variable = "${module.production_cluster.oidc_provider}:aud"
-      values   = ["sts.amazonaws.com"]
-    }
-
-    principals {
-      type        = "Federated"
-      identifiers = [module.production_cluster.oidc_provider_arn]
-    }
-  }
-}
-
-resource "aws_iam_role" "cache_indexer" {
-  name               = "CacheIndexer"
-  assume_role_policy = data.aws_iam_policy_document.cache_indexer_assume_role_policy.json
-}
-
-data "aws_iam_policy_document" "cache_indexer_policy" {
-  statement {
-    sid     = ""
-    effect  = "Allow"
-    actions = ["s3:GetObject"]
-
-    resources = [
-      "arn:aws:s3:::spack-binaries/*"
-    ]
-  }
-
-  statement {
-    sid     = ""
-    effect  = "Allow"
-    actions = ["s3:PutObject", "s3:DeleteObject"]
-
-    resources = [
-      "arn:aws:s3:::spack-binaries/cache_spack_io_index.json"
-    ]
-  }
+moved {
+  from = aws_iam_role.cache_indexer
+  to   = module.production_cluster.aws_iam_role.cache_indexer
```

Comment:
How are you going to handle these moved blocks? Merge this PR, run `terraform apply`, and then make a follow up PR removing them?

Reply:
Yes

Reply:
:+1: